Source/TCPServer.swift

```diff
@@ -0,0 +1,71 @@
+// TCPServer.swift
+//
+// The MIT License (MIT)
+//
+// Copyright (c) 2015 Zewo
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import CLibvenice
+import C7
+
+public final class TCPServer {
+    
+    public var uri: C7.URI
+    private let backlog: Int32
+    private let reusePort: Bool
+    private var socket: tcpsock?
+    private var connection: TCPConnection?
+    
+    public init(for uri: C7.URI) throws {
+        self.uri = uri
+        backlog = 128
+        reusePort = false
+    }
+    
+    public init(for uri: URI, backlog: Int = 128, reusePort: Bool = false) throws {
```

Comment:
`public init(for uri: URI, queuing backlog: Int = 128, reusingPort reusePort: Bool = false) throws`